src/platform/packages/shared/kbn-lens-embeddable-utils/config_builder/transforms/charts/xy/chart.ts

```diff
@@ -0,0 +1,439 @@
+  | 'minBarHeight'
+  | 'hideEndzones'
+  | 'showCurrentTimeMarker'
+  | 'pointVisibility'
+> {
+  return {
+    ...(config.decorations?.value_labels != null
+      ? { valueLabels: config.decorations?.value_labels ? 'show' : 'hide' }
+      : {}),
+    ...(config.decorations?.line_interpolation
+      ? { curveType: curveType[config.decorations?.line_interpolation] }
+      : {}),
+    ...(config.decorations?.fill_opacity != null
+      ? { fillOpacity: config.decorations?.fill_opacity }
+      : {}),
+    ...(config.decorations?.minimum_bar_height
+      ? { minBarHeight: config.decorations?.minimum_bar_height }
+      : {}),
+  };
+}
+
+type LayerToDataView = Record<string, string>;
+
+function buildLayers(layers: XYState['layers'], dataViews: LayerToDataView): XYLensState['layers'] {
+  return layers
+    .map((layer, index) => buildXYLayer(layer, index, dataViews[getIdForLayer(layer, index)]))
+    .filter(nonNullable);
+}
+
+export function buildVisualizationState(
+  config: XYState,
+  usedDataViews: LayerToDataView
+): XYLensState {
+  const layers = buildLayers(config.layers, usedDataViews);
+  return {
+    preferredSeriesType: layers.filter(isLensStateDataLayer)[0]?.seriesType ?? 'bar_stacked',
+    ...convertLegendToStateFormat(config.legend),
+    ...convertFittingToStateFormat(config.fitting),
+    ...convertAxisSettingsToStateFormat(config.axis),
+    ...convertAppearanceToStateFormat(config),
+    layers,
+  };
+}
+
+export function buildVisualizationAPI(
+  config: XYLensState,
+  layers: Record<string, DataSourceStateLayer>,
+  adHocDataViews: Record<string, unknown>,
+  references: SavedObjectReference[],
+  internalReferences: SavedObjectReference[]
+): XYState {
+  const dataLayers = config.layers.filter(isLensStateDataLayer);
+  if (!dataLayers.length) {
+    throw new Error('At least one data layer is required to build the XY API state');
+  }
+  if (dataLayers.every((layer) => layer.accessors.length === 0)) {
+    throw new Error(
+      'Data layers must have at least one accessor defined to build the XY API state'
+    );
+  }
+  return {
+    type: 'xy',
+    ...convertLegendToAPIFormat(config.legend),
+    ...convertFittingToAPIFormat(config),
+    ...convertAxisSettingsToAPIFormat(config),
+    ...convertAppearanceToAPIFormat(config),
+    layers: buildXYLayerAPI(config, layers, adHocDataViews, references, internalReferences),
+  };
+}
+
+function convertFittingToAPIFormat(config: XYLensState): Pick<XYState, 'fitting'> | {} {
+  const fittingOptions = {
+    ...(config.fittingFunction ? { type: config.fittingFunction } : {}),
+    ...(config.emphasizeFitting ? { dotted: config.emphasizeFitting } : {}),
+    ...(config.endValue ? { endValue: config.endValue } : {}),
+  };
+
+  if (Object.keys(fittingOptions).length === 0) {
+    return {};
+  }
+
+  return { fitting: fittingOptions };
+}
+
+function convertExtendsToAPIFormat(
+  extent: AxisExtentConfig | undefined
+): { extent: ExtentsType } | {} {
```

Comment:
```suggestion
): { extent?: ExtentsType } {
```

In this way you don't need to add a type assertion where this fn is used

Reply:
Doesn't seem to work.
I'll try to drop the type casting below, but that optional thing is not the root cause of it.